lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -344,24 +348,41 @@ void SessionStateEnforcer::m5g_update_session_state_to_amf(
                  << " with teid " << teid;
     return;
   }
+  bool amf_update_pending = false;
+  uint32_t cur_version;
   auto& session       = **session_it;
   auto session_update = SessionStore::get_default_session_update(session_map);
   SessionStateUpdateCriteria& session_uc =
       session_update[imsi][session->get_session_id()];
-  session->set_fsm_state(new_state, session_uc);
-  uint32_t cur_version = session->get_current_version();
-  session->set_current_version(++cur_version, session_uc);
-  bool update_success = session_store_.update_sessions(session_update);
-  if (update_success) {
-    MLOG(MINFO) << "Updated SessionStore SessionState based on UPF message "
+  switch  (session->get_state()) {
+     case CREATED:
+        session->set_fsm_state(ACTIVE, session_uc);
+        cur_version = session->get_current_version();
+        session->set_current_version(++cur_version, session_uc);
+	amf_update_pending = true;
+	break;
+     case RELEASE:
+	m5g_complete_termination(session_map,
+		imsi,session->get_session_id(),session_update);
+     default:
+	 break;
+  }
+  if (amf_update_pending) {
+    bool update_success = session_store_.update_sessions(session_update);
+    if (update_success) {
+       MLOG(MINFO) << "Updated SessionStore SessionState based on UPF message "
                 << " with session_id" << session->get_session_id();
-  } else {
-    MLOG(MERROR) << "Failed to update SessionStore based on UPF message"
+    } else {
+       MLOG(MERROR) << "Failed to update SessionStore based on UPF message"
                  << " with session_id" << session->get_session_id();
+    }
+    /* Update the state change notification to AMF */
+    handle_state_update_to_amf(
+        imsi, *session, magma::lte::M5GSMCause::OPERATION_SUCCESS);
+  }
+  else {
+    session_store_.update_sessions(session_update);

```

Comment:
i think you need to run format
`make precommit_sm`

Reply:
Ran, resolved.